lib/i18n_sonder/workers/sync_translations.rb

```diff
@@ -19,17 +22,14 @@ def sync(approved_translations_only:)
           #   }
           # }
           if approved_translations_only
-            translation_result = localization_provider.approved_translations(language.to_s)
+            localization_provider.approved_translations(language.to_s, BATCH_SIZE) do |translation_result_batch|
+              successful_syncs = process_translation_result(translation_result_batch, language, successful_syncs)
+            end
```

Comment:
do we have to do any error handling if the batch being processed fails?

Reply:
no, the `translation_result_batch` contains a `success` and `failure` object that captures failures for individual files. That's why we handle them separately in `process_translation_result`.